# Import Statements

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
c_df = pd.read_csv('ClimateData.csv')
c_df.head()

id building_class  floor_area  year_built  precipitation_inches  \
0   0     Commercial     61242.0      1942.0                 16.59   
1   1     Commercial    274000.0      1955.0                 16.59   
2   2     Commercial    280025.0      1951.0                 16.59   
3   3     Commercial     55325.0      1980.0                 16.59   
4   4     Commercial     66000.0      1985.0                 16.59   

   snowfall_inches   avg_temp  days_with_fog  site_eui  
0              0.0  56.972603            NaN    248.68  
1              0.0  56.972603           12.0     26.50  
2              0.0  56.972603           12.0     24.69  
3              0.0  56.972603           12.0     48.41  
4              0.0  56.972603            NaN      3.90

# Data Understanding and Summary

In [4]:
c_df.columns

Index(['id', 'building_class', 'floor_area', 'year_built',
       'precipitation_inches', 'snowfall_inches', 'avg_temp', 'days_with_fog',
       'site_eui'],
      dtype='object')

In [5]:
print(c_df.dtypes)

id                        int64
building_class           object
floor_area              float64
year_built              float64
precipitation_inches    float64
snowfall_inches         float64
avg_temp                float64
days_with_fog           float64
site_eui                float64
dtype: object


# Unique Value Analysis

In [6]:
print(c_df['building_class'].unique())

['Commercial' 'Residential']


# Descriptive Statistics

In [7]:
c_df.describe()

id    floor_area    year_built  precipitation_inches  \
count  75757.000000  7.575700e+04  73920.000000          75757.000000   
mean   37878.000000  1.659839e+05   1952.306764             42.430651   
std    21869.306509  2.468758e+05     37.053619             10.592205   
min        0.000000  9.430000e+02      0.000000              0.000000   
25%    18939.000000  6.237900e+04   1927.000000             39.350000   
50%    37878.000000  9.136700e+04   1951.000000             42.170000   
75%    56817.000000  1.660000e+05   1977.000000             46.320000   
max    75756.000000  6.385382e+06   2015.000000            107.690000   

       snowfall_inches      avg_temp  days_with_fog      site_eui  
count     75757.000000  75757.000000   29961.000000  75757.000000  
mean         29.136379     56.176705     109.142051     82.584707  
std          17.810894      2.241940      50.699751     58.255413  
min           0.000000     44.512329      12.000000      1.000000  
25%           9.600000     55.147541      88.000000     54.530000  
50%          31.600000     56.835616     104.000000     75.290000  
75%          49.100000     57.215847     131.000000     97.280000  
max         127.300000     77.245205     311.000000    997.870000

# Missing Values Assessment

In [8]:
invalid_entries=c_df[c_df.select_dtypes(include=['number'])<0].count()
print("Invalid Entries:\n",invalid_entries)

Invalid Entries:
 id                      0
building_class          0
floor_area              0
year_built              0
precipitation_inches    0
snowfall_inches         0
avg_temp                0
days_with_fog           0
site_eui                0
dtype: int64


In [10]:
missing_df=c_df.isnull().sum()
print('Missing data:\n',missing_df)

Missing data:
 id                          0
building_class              0
floor_area                  0
year_built               1837
precipitation_inches        0
snowfall_inches             0
avg_temp                    0
days_with_fog           45796
site_eui                    0
dtype: int64


There looks to be approximately 1837 missing entries in the year_built column and 45,796 entries missing in the days_with_fog column

# Missing Data Handling

There are multiple ways to handle the missing data in the dataset, so we'll explore two of them here

# Duplicate Records

# Outlier Detection with Boxplots

# Histogram Analysis for Distribution

# Feature Scaling